### DISCUSSION TITLE
Extra metadata in a field schema

### DISCUSSION THREAD

**Discussion by koltyakov** [pocketbase/pocketbase#3590] [Q&A]
Are there any plans for adding extra metadata on field level, such as Display Name and Description. It could be handy for prototyping, auto documentation and client applications being able retrieving such metadata from the entity model.

**Comment by ganigeorgiev** [comment]
I can't think of a scenario where a client-side application may need a field label (_how translations will be handled?_) but for now there are no plans for this as in my opinion it will clutter the Admin UI without adding too much benefit.

**Reply by koltyakov** [reply]
I'd expect many projects with PocketBase start with a single language. I'm not touching translations here as it could be a lot of work. The strategy for when and if a localizations feature is considered as a part of the core: a value from Display Name can go to a default translation.

Currently, modeling an entity schema one got to maintain separate documentation with something like:

## Entity

Display Name | Internal Name | Data Type | Constraint | Description
---|---|---|---|---
Title | `title` | `string(50)` | required | This is a field storing ...

Having this right in the entity editor will allow auto-documenting, avoid docs-model sync issues. And for the majority of projects provide the mapping for human readable names.

It should not oversaturate the interface a lot, as a field configuration is hidden behind the cog button anyways.

E.g. if only adding Display Name:

<img width="663" alt="image" src="https://github.com/pocketbase/pocketbase/assets/7816483/b10d8f98-e5f6-47fb-853e-5be6c46475fc">

vs (as is)

<img width="652" alt="image" src="https://github.com/pocketbase/pocketbase/assets/7816483/e1dace5b-ad7e-495e-9d22-3e4971590799">

**Reply by ganigeorgiev** [reply]
Depending on the users demand I may reconsider it in the future, but for now this remains out of the scope for v1.

**Reply by koltyakov** [reply]
Thank you for the answers!

**Reply by koltyakov** [reply]
@ganigeorgiev if it's not a principal disagreement adding Display Name please check the draft PR https://github.com/pocketbase/pocketbase/pull/3591

Anyways, it waits for "users demand". Just a concept of a change which would be great to introduce.
